src/Text/LLVM/AST.hs

```diff
@@ -841,6 +875,22 @@ data Visibility = DefaultVisibility
                 | ProtectedVisibility
     deriving (Data, Eq, Generic, Ord, Show, Typeable)
 
+data ThreadLocality = NotThreadLocal
+                | ThreadLocal
+                | LocalDynamic
+                | InitialExec
+                | LocalExec
+    deriving (Data, Eq, Generic, Ord, Show, Typeable)
+
+-- ^ there is also None, use Nothing for it
+data UnnamedAddr = GlobalUnnamedAddr
+                 | LocalUnnamedAddr
+    deriving (Data, Eq, Generic, Ord, Show, Typeable)
+
+newtype AddrSpace = AddrSpace
```

Comment:
The comments in [this part](https://llvm.org/docs/LangRef.html#data-layout) of the LLVM Language Reference Manual suggest that an address space's value must be less than `2^24`. Is that worth putting in the Haddocks here?